Sources/Shared/Combine/VimeoClient+Publisher.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  VimeoClient+Publisher.swift
+//  VimeoNetworking
+//
+//  Created by Rogerio de Paula Assis on 11/17/19.
+//  Copyright © 2019 Vimeo. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+extension VimeoClient {
+
+    /// Returns a publisher that wraps a decodable request for a given URLConvertible type.
+    /// - Parameter request: the request convertible for which to create a decodable request.
+    ///
+    /// The publisher publishes the specified decodable type when the request completes and immediately
+    /// sends a completion event. If the request fails, it terminates with an error.
+    @available(iOS 13, tvOS 13, macOS 10.15, *)
+    func publisher<D: Decodable>(for endpoint: EndpointType) -> AnyPublisher<D, Error> {
+        let passthroughSubject = PassthroughSubject<D, Error>()
+        var task: RequestToken?
+        task = self.request(endpoint) { (result: Result<D, Error>) in
+            switch result {
+            case .success(let decodable):
```

Comment:
If we're making a request to an endpoint but not expecting an object back, as is the case for a `VIMNullResponse`, would this still work? I guess we'd just need an empty object that conforms to `Decodable` and ignore the result?